ChaosMod/Effects/db/Peds/PedsJesusRain.cpp

```diff
@@ -0,0 +1,110 @@
+#include <stdafx.h>
+
+//Effect by ProfessorBiddle
+
+static void OnTick()
+{
+	static const Hash parachute = GET_HASH_KEY("GADGET_PARACHUTE");
+	static const Hash playerGroup = GET_HASH_KEY("PLAYER");
+	static const Hash civGroup = GET_HASH_KEY("CIVMALE");
+	static const Hash femCivGroup = GET_HASH_KEY("CIVFEMALE");
+	static const std::vector<Hash>& weps = Memory::GetAllWeapons();
+	static constexpr int maxJesus = 40;
+	static constexpr Hash jesusHash = -835930287;
+	static Ped jesuss[maxJesus] = {};
+	static int jesusDespawnTime[maxJesus];
+	static int jesusAmount = 0;
+	Ped playerPed = PLAYER_PED_ID();
+	Vector3 playerPos = GET_ENTITY_COORDS(playerPed, false);
+	static DWORD64 lastTick = 0;
+	DWORD64 curTick = GetTickCount64();
+
+	if (jesusAmount <= maxJesus && curTick > lastTick + 800)
+	{
+		lastTick = curTick;
+		Vector3 spawnPos;
+		spawnPos.x = playerPos.x + g_random.GetRandomInt(-20, 20);
+		spawnPos.y = playerPos.y + g_random.GetRandomInt(-20, 20);
+		spawnPos.z = playerPos.z + g_random.GetRandomInt(50, 75);
+		LoadModel(jesusHash);
+		Ped jesus = CREATE_PED(28, jesusHash, spawnPos.x, spawnPos.y, spawnPos.z, g_random.GetRandomInt(0, 359), true, false);
+		jesusAmount++; 
+		for (int i = 0; i < maxJesus; i++)
+		{
+			Ped& ped = jesuss[i];
+			if (!ped)
+			{
+				ped = jesus;
+				jesusDespawnTime[i] = 5;
+				break;
+			}
+		}	
+		SET_MODEL_AS_NO_LONGER_NEEDED(maxJesus);
+		//Give em' a shove
+		APPLY_FORCE_TO_ENTITY_CENTER_OF_MASS(jesus, 0, 35.f, 0, -2500.f, true, false, true, true);
+		//give parachute, for some reason only works with both
+		SET_PED_GADGET(jesus, parachute, true);
+		GIVE_WEAPON_TO_PED(jesus, parachute, 9999, true, true);
+		//Jesus hates criminals
+		Hash relationshipGroup;
+		ADD_RELATIONSHIP_GROUP("_HOSTILE_JESUS", &relationshipGroup);
+		SET_RELATIONSHIP_BETWEEN_GROUPS(5, relationshipGroup, playerGroup);
+		SET_RELATIONSHIP_BETWEEN_GROUPS(5, relationshipGroup, civGroup);
+		SET_RELATIONSHIP_BETWEEN_GROUPS(5, relationshipGroup, femCivGroup);
+		SET_PED_RELATIONSHIP_GROUP_HASH(jesus, relationshipGroup);
+		//random weapons
+		Hash wep = weps[g_random.GetRandomInt(0, weps.size() - 1)];
+		GIVE_WEAPON_TO_PED(jesus, wep, 9999, true, false);
+		SET_CURRENT_PED_WEAPON(jesus, wep, true);
+		//Make jesus badass, but not as badass as Griefer Jesus		
+		SET_ENTITY_PROOFS(jesus, false, true, true, true, false, false, false, false);		
+		SET_PED_CAN_RAGDOLL_FROM_PLAYER_IMPACT(jesus, true);
+		SET_RAGDOLL_BLOCKING_FLAGS(jesus, 5);
+		SET_PED_SUFFERS_CRITICAL_HITS(jesus, true);
+		//Safely parachute to player (sometimes)
```

Comment:
Shouldn’t they all land safely?

Reply:
I can't figure out how to make them all land safely, they should but some of them always remove their parachutes and fall